### PR TITLE
Apply review feedback

### PR DESCRIPTION
**Pull request by alpe**
Contains
* Cleanup
* Better comments
* Integration tests with real wasm contract

Note: the ibc-reflect contract used is https://github.com/CosmWasm/cosmwasm/pull/1685

### CODE REVIEW

File: x/wasm/keeper/relay.go [hunk 1]
```diff
@@ -133,7 +133,10 @@ func (k Keeper) OnRecvPacket(
 	res, gasUsed, execErr := k.wasmVM.IBCPacketReceive(codeInfo.CodeHash, env, msg, prefixStore, cosmwasmAPI, querier, ctx.GasMeter(), gas, costJSONDeserialization)
 	k.consumeRuntimeGas(ctx, gasUsed)
 	if execErr != nil {
-		panic(execErr) // let contract fully abort IBC receive in certain case
+		panic(execErr) // let the contract fully abort an IBC packet receive.
```
Comment (ethanfrey) [inline x/wasm/keeper/relay.go:136]: 👍

**Review by ethanfrey — APPROVED** [review]
Good stuff. Why aren't they done inline in the PR itself? (I can review the diff or a commit or two)

I guess the e2e tests deserve their own PR. But fine to merge this and mark some comments as resolved.

**Comment by alpe** [comment]
Thanks for the feedback!
> Why aren't they done inline in the PR itself?

I need to backport the changes to main. I hope this is easier with a separate PR.

**Comment by alpe** [comment]
@Mergifyio backport main

**Comment by alpe** [comment]
@mergifyio backport main

**Pull request merged**